include/xten/Dialect/XTenNN/IR/XTenNNOps.td

```diff
@@ -166,6 +166,72 @@ def XTenNN_DequantizeOp: XTenNN_Op<"dequantize", [
   let assemblyFormat = [{ `(`$input `:` type($input)`)` attr-dict `->` type($output) }];
 }
 
+def XTenNN_GroupQuantizeOp: XTenNN_Op<"group_quantize", [
+            Pure]> {
+  let summary = "Group quantizes a float32 tensor to an integer tensor of given width.";
+  let description = [{
+    Quantizes a given float32 tensor into a integer tensor of given width.
+
+    Scales and zeros must have a shape that is broadcastable to the input shape.
+    Input and scales most have the same dtype, as must zeros and quants.
+    The range of values in zeros and quants is given by the min and max attributes,
+    and the bits attribute gives the number of integer bits required for that range.
+
+    Applies the following linear quantization:
+      quants[a][b][c] = round( (input[a][b][c] / scales[a][b][c]) + zeros[a][b][c] )
```

Comment:
Does that mean this is an elementwise operation? I assume a, b, c are indices.

Reply:
Yes, once `scales` and `zeros` are broadcast.

Reply:
Would the ElementwiseTrait make sense for the operation then?

Reply:
I couldn't work out how to use that when I depend on broadcast. If you know, can you contribute in followup?